static/js/teachers.ts

```diff
@@ -399,116 +382,93 @@ export function save_customizations(class_id: string) {
-                  </div>
-                  `
-                  $(this).append(div);
-                }
-                drag_list(document.getElementById("level-"+level));
-            });
-            for (let i = 1; i <= 18; i++) {
-              available_adventures[i] = [];
-            }
-            for (let i = 0; i < teacher_adventures.length; i++) {
-              available_adventures[teacher_adventures![i]['level']].push({'name': teacher_adventures[i]['id'], 'from_teacher': true});
-            }
-            modal.notifySuccess(response.success);
-        }).fail(function (err) {
-            modal.notifyError(err.responseText);
-        });
+      // We need to know the current level that is selected by the user
+      // so we can know which level to draw in the template  
+      let active_level_id : string = $('[id^=level-]')[0].id;
+      let active_level = active_level_id.split('-')[1]
+      htmx.ajax(
+        'POST',
+        `/for-teachers/restore-customizations?level=${active_level}`,
+        '#adventure-dragger'
+      ).then(() => {
+        // Restore all the options other than the adventures.
+        // The adventures will be restored to the default using an HTMX call to the server
+        $('.other_settings_checkbox').prop('checked', false);
+        // Remove the value from all input fields -> reset to text to show placeholder
+        $('.opening_date_input').prop("type", "text")
+                                .blur()
+                                .val('')
+                                .prop('disabled', false)
+                                .attr('placeholder', ClientMessages.directly_available)
+                                .attr('placeholder', ClientMessages.directly_available)
+                                .each(function() {         
+                                      if($(this).hasClass('bg-green-300')) {
+                                        $(this).removeClass('bg-green-300')
+                                              .addClass('bg-gray-200')
+                                      }
+                                });
+
+        $('[id^=enable_level_]').prop('checked', true);                
+        setLevelStateIndicator(active_level);
+        modal.notifySuccess(ClientMessages.customization_deleted);          
+      })
     });
 }
 
-export function select_all_levels_adventure(adventure_name: string) {
-    markUnsavedChanges();
-    let first_input = true;
-    let checked = true;
-    $('.adventure_level_input').each(function() {
-        const name = <string>$(this).attr('adventure');
-        if (name == adventure_name && $(this).is(":visible")) {
-            if (first_input) {
-                checked = $(this).prop("checked");
-                first_input = false;
-            }
-            $(this).prop("checked", !checked);
-        }
-    });
+export function restore_adventures_to_default(prompt: string) {
```

Comment:
I'm using this function to fire the htmx call from the button, which will in return call the function from python. I do it this way since we need to get the current level that the user is editing, so when they restore the button won't be kicked out to level 1. Also, because we need to display the modal.

Reply:
Aha, I see!

> I'm using this function to fire the htmx call from the button

I would say that ideally this should not be necessary. The "htmx way" to do this would be to have a button like this:

```
<button hx-post="/teacher-adventures/{{ level }}/restore">
```

Now I'm guessing you're probably doing this with JavaScript because I think your page layout might look something like this:

![image](https://github.com/hedyorg/hedy/assets/524162/8d2261a5-1122-4ea4-9006-38f04dae9c2c)

So, you need some external mechanism to transfer the level number from the selector to the reset button in order to trigger the appropriate reset.

------------

However, if you put the button *inside* the HTML fragment that HTMX renders, it's all a lot easier! When loading the HTML fragment that will go into the orange box, we *know* that we are rending `/adventures/5`, so we *know* that `{{ level }} == 5`, so we can immediately substitute that into the URL for the reset button:

![image](https://github.com/hedyorg/hedy/assets/524162/e416a1d9-058e-4bcb-8c99-96f6edb211b2)

Here is the trick: you render the buttons that take actions on data along with the data you are rendering! That way, you know all the identifiers you need to know in order to generate the correct URLs.

Reply:
The above pattern holds if you need to replace the entire content area, for example for a button like "reset". 

If you have a list of items you want to be able to edit, you could do the same for a "save all" button, or you could have individual "saves" (automatic or via a button click) for individual elements. Both are fine patterns.

In both cases you'd have a `<form>` to hold the data you want to send to the server, and in both cases you'd probably re-render the exact same page after saving, with potentially a "Saved" announcement rendered in as well.

![image](https://github.com/hedyorg/hedy/assets/524162/d02bd017-1606-459a-8c47-246fe6ba1065)